### DISCUSSION TITLE
sumatra opens the same files multiple times

### DISCUSSION THREAD

**Discussion by claudiotam** [sumatrapdfreader/sumatrapdf#3889] [Q&A]
Hello!
I have a problem with sumatra on win11 in disabled tabs mode.
I double click on file1, and sumatra correctly opens file1.
Then I double click on file2.
Sumatra opens file1 again (duplicate windows) and file2.
Then I double click on file3.
Sumatra opens file1 and file2 again (duplicate windows) and file3.
Is this normal? How do I correct this strange behaviour? I tried other pdf readers and they don't behave like this.

**Comment by claudiotam** [comment]
If I put option "remember opened files" off, then it behaves normally (one double-click one window). But it also forgets page and zoom of every pdf file, so I would like to keep this option on.

**Reply by GitHubRulesOK** [reply]
I think this should have been fixed in current pre-release https://www.sumatrapdfreader.org/prerelease

**Reply by claudiotam** [reply]
the problem also appears in the current pre-release

**Reply by MarcoNovaro** [reply]
I have the same issue as well (Windows 10). Currently I'm working with "remember opened files" OFF, but I'd like to see a fix for this problem...

**Comment by mayakot98** [comment]
I am having the same problem on win 11. I handled the issue by using older 3.4.6 version from https://www.sumatrapdfreader.org/download-prev

**Comment by kjk** [comment]
This should be fixed in latest pre-release version https://www.sumatrapdfreader.org/prerelease

**Reply by mayakot98** [reply]
I have already check the Pre-Release. Problem Still persists

**Reply by claudiotam** [reply]
Problem persists also in the pre-release; same behaviour of the OP.

**Reply by kjk** [reply]
Which version of pre-release? If it's not 15900, then check again.

If it still happens, then provide:
- step-by-step instructions to repro
- do you have tabs enabled or disabled?
- what is `ReuseInstance` advanced setting?
- when you reproduce it, use `Ctrl + K`, `Show Log` and attach the log

**Comment by kjk** [comment]
Setting advanced setting `ReuseInstance = true` should fix it.

I thought I've added a work-around even when `ReuseInstance = false` but seems there's a bug there.

**Reply by MarcoNovaro** [reply]
This is not working for me, even with the latest prerelease

**Reply by kjk** [reply]
Do you have `ReuseInstance = true` ?

**Reply by MarcoNovaro** [reply]
Yes, I set it in the advanced settings

**Reply by mayakot98** [reply]
![Screenshot_1](https://github.com/sumatrapdfreader/sumatrapdf/assets/102904701/efd1ad79-c3ea-4776-a432-d045709d4de4)

These are my settings, but still no solution yet

**Reply by mayakot98** [reply]
https://github.com/sumatrapdfreader/sumatrapdf/assets/102904701/c03ad7b4-5944-4664-8b2a-157a44c56448

**Reply by kjk** [reply]
@mayakot98 What you show is expected behavior.

You have multiple files open and you close Sumatra. We save the session.

You double click on a file which opens the file and also restores the 3 files from session because that's the expected behavior for most people.

You can set `RestoreSession = false` to disable restoring the session but it'll disable it always.

**Reply by mayakot98** [reply]
from above solution my one problem is solved but still it always open home tab along with the file.

please suggest any solution . below is a short demo


https://github.com/sumatrapdfreader/sumatrapdf/assets/102904701/7d4d8c65-6ded-4929-bba1-11808af61d45

**Reply by GitHubRulesOK** [reply]
@mayakot98 
home tab is an advanced setting `NoHomeTab = true` should do that

**Reply by mayakot98** [reply]
Thanks it solves my problem.

**Comment by vrubleg** [comment]
I disable tabs, so all documents are open in separate windows. Whenever I open a document, SumatraPDF v3.5.2 also opens another window with the previously closed document. So instead of 1 window with one document that I just opened I get 2 windows. Is it the same issue or another one?

**Reply by vrubleg** [reply]
OK, disabling "remember opened files" fixes the issue. But this setting makes zero sense for mode without tabs, it should be disabled by default.

**Reply by kjk** [reply]
If you set `ReuseInstance = true` advanced setting it should fix it.
Or upgrade to latest pre-release https://www.sumatrapdfreader.org/prerelease

**Comment by claudiotam** [comment]
These are the default values, after installing a fresh version of sumatra and setting `use tabs = false`.

default values for v3.5.2
```
restore session = true
reuse instance = false
```

default values for v3.6.15900
```
restore session = true
reuse instance = true
```

I've tried all 8 combinations with these findings.
- The version does not change the outcome.
- The option `reuse instance` does not change the outcome.
- The option `restore session` always doesn't work if set to true; always works if set to false.

So, I just use `restore session = false` and the bug is solved.

NOTE: to me, it makes no sense to have `use tabs = false` and `restore session = true` at the same time; and this is an almost default configuration. This is just a note, the problem is solved.

**Reply by GitHubRulesOK** [reply]
@claudiotam Just a comment that, the common use for tab-less and restore pairs of files as a session was as far as I recall mainly for side by side viewing before tabs were added and with tabs before tear off tabs or new window)

Where-as reuse instance was originally needed for LaTeX previewing, where one tab-less window needed constant reuse (no fresh windows on reload). When tabs arrived the suggestion was use tabs with change reuse to true (as preferable, now the default?)

Restore session is for multiple windows or now tabs to re-appear on restart as groups

**Comment by Back-T0** [comment]
Whenever I try to open a file twice in Sumatra, it keeps pointing to the one that's already open. Is there a way to fix this through the advanced settings?

**Reply by Back-T0** [reply]
I want to do this because sometimes I need to compare with what was said earlier, and I think there are others who feel the same way.

**Reply by GitHubRulesOK** [reply]
@Back-T0
 see my comment above the control for reuse is `ReuseInstance = true` 
so if its set to true then try with it set to false (close and restart SumatraPDF to test how it is working. however see below.

IF I have tabs OFF I get 2 windows for same file OR I can duplicate with CTRL + SHIFT + N
With TABS ON I dont get a second window so only CTRL + SHIFT + N will start a second window with same file OR use `-new-window` on command line

**Reply by kjk** [reply]
Assuming you use tabs.

If you drag&drop, it'll open in a new tab.

You can use `Ctrl + K` command `Open Current Document In New Window`.

**Reply by kjk** [reply]
Also, if you use `Ctrl + K` and select a file from the list of recently opened files at the top, it opens in a new tab

**Reply by kjk** [reply]
in next pre-release if you press `Ctrl` while double-clicking to open, should always open in a new tab (vs. activating a tab if already loaded).

**Reply by Back-T0** [reply]
This is undoubtedly the better option

**Comment by Mr-Rayman** [comment]
I do have the same problem in current release 3.6.1 (64 bit).  This is very annoying, all previous files are opened in separate windows when just double-clicking one PDF from the filesystem (I don't use the tabbed view).

Btw: This works as workaround:
restore session = false
